pmlc/dialect/op_lib/cpp/WrapperGen.cc

```diff
@@ -0,0 +1,162 @@
+//===- WrapperGen.cc - MLIR op lib dialect wrapper generator for C++ ------===//
+//
+// Copyright 2019 Intel Corporation.
+//
+// =============================================================================
+//
+// WrapperGen in C++ generates a file that wraps the C++ op lib with a fluent
+// interface.
+//
+//===----------------------------------------------------------------------===//
+
+#include <regex>
+#include <vector>
+
+#include "mlir/TableGen/Format.h"
+
+#include "pmlc/dialect/op_lib/cpp/WrapperGen.h"
+#include "pmlc/dialect/op_lib/cpp/utils.h"
+
+using llvm::raw_ostream;
+using llvm::RecordKeeper;
+
+namespace pmlc::dialect::op::tblgen::cpp {
+
+namespace wrapper {
+
+OpEmitter::OpEmitter(const OpInfo& op, raw_ostream& os) : opInfo_(OpInfo(op)) {
+  os << formatv(commentHeader, opInfo_.name_, "wrapper");
+  os << "class " << opInfo_.name_ << " {\n";
+  os.indent(2) << "protected:\n";
```

Comment:
This probably should be `private:` (I don't know if we need to support having subclasses)